core/src/main/scala/org/bitcoins/core/protocol/dlc/compute/DLCAdaptorPointComputer.scala

```diff
@@ -50,6 +45,70 @@ object DLCAdaptorPointComputer {
     nonce.add(pubKey.publicKey.tweakMultiply(FieldElement(hash)))
   }
 
+  /** This trie is used for computing adaptor points for a single oracle corresponding
+    * to digit prefixes while memoizing partial sums.
+    *
+    * For example the point corresponding to 0110 and 01111010 both begin with
+    * the 011 sub-sum.
+    *
+    * This trie stores all already computed sub-sums and new points are computed
+    * by extending this Trie.
+    *
+    * Note that this method should not be used if you have access to LibSecp256k1CryptoRuntime
+    * because calling CryptoUtil.combinePubKeys will outperform memoization in that case.
+    */
+  case class AdditionTrieNode(
+      preComputeTable: Vector[Vector[ECPublicKey]], // Nonce -> Outcome -> Point
+      depth: Int = 0,
+      var children: Vector[AdditionTrieNode] = Vector.empty,
```

Comment:
aren't these global mutable state?

Reply:
```scala
  private case class AdditionTrieNode(
      preComputeTable: Vector[Vector[ECPublicKey]], // Nonce -> Outcome -> Point
      depth: Int = 0,
      private var children: Vector[AdditionTrieNode] = Vector.empty,
      private var pointOpt: Option[SecpPoint] = None) {

```

Reply:
Is this a binary trie? If so can we add the invariant `require(children.length < 3)`

Reply:
Good point on the `private`s :P Don't know how I missed that (fixed).

Re binary trie: the current implementation of the Trie is base agnostic so that it works with any base. It still has the invariant however that `children.length <= base` but this can't actually be enforced by a `require` since `children` is mutable and always starts out as `Vector.empty` so this `require` would trivially pass.